File: problem-two.js
// Let's do generators in JS too since we did it in Python

function* fibonacci() {
  let current = a = 1
  let b = 2

  while (true) {
    current = b

    yield a

    b = a + current
    a = current
  }
}

const main = () => {
  const f = fibonacci()
  let lastNumber = 0
  let fibSum = 0

  while (lastNumber < 4e6) {
    lastNumber = f.next().value
    fibSum += lastNumber % 2 === 0 ? lastNumber : 0
  }

  console.log(fibSum)
  return fibSum
}

require.main === module && main()

File: problem-two.py
a, b = 0, 1

def fibonacci():
    global a, b
    while True:
        a, b = b, a + b
        yield a

def main():
    f = fibonacci()
    last_number = 0
    fib_sum = 0

    while last_number < 4e6:
        last_number = next(f)
        if last_number % 2 == 0:
            fib_sum += last_number

    print(fib_sum)

if __name__ == "__main__":
    main()

